Added account page

TwiN opened the pull request:
This PR adds a simple account page that gets the data from the bot's API.

![account page](https://twinnation.org/api/screenshots/2018-06-05_145909.png)

Obviously, this is a simple implementation, but it could help reduce requests by minimizing the use of the `!balance` commands.

For now, I didn't actually create a separate HTML page because I felt it was unnecessary.

I also fixed a few minor derps (e.g. logo not centered,  `bugs` link not opening in new tab, etc.)

Let me know what you think.

-----------

**UPDATE:** Bot went down the moment I created this PR, so testing it will have to wait until it's up again.

TwiN committed:
Centered logo in nav

TwiN committed:
Centered logo in nav

TwiN committed:
Started working on adding account page

TwiN committed:
Started working on adding account page

TwiN committed:
Make nav links open in new page (e.g. GitHub link)

TwiN committed:
Make nav links open in new page (e.g. GitHub link)

TwiN committed:
Finished simple account page implementation

TwiN committed:
Finished simple account page implementation

dzervas committed:
[docker] Minor fix in docker-compose

dzervas committed:
[docs] Stopped multiplying investments with 10k

jimbobur committed:
fixed win/lose/nothing if statments

thecsw commented:
OOOh, very nice! If everything is ok, some dev will merge this. Thanks!

robalb committed:
added apocalipse warning popup

robalb committed:
Merge branch 'master' of https://github.com/MemeInvestor/memeinvestor_bot

thecsw commented:
Do you have some github.io page or something where I can take a look?

P.S. We have some merge conflicts! Please resolve them and send a testing webpage.

TwiN committed:
Merged changes from master

TwiN committed:
Merge remote-tracking branch 'origin/patch-69' into patch-69

# Conflicts:
#	docs/index.html

TwiN commented:
That was one hell of a conflict. My head hurts.

TwiN commented:
@thecsw you can just open `index.html` with a browser. Since it uses https://memes.market/api, it should work even without the bot running on your local machine.

But basically, all I did for the account 'page' is adding a container at the end of `index.html` that is hidden unless the "account" navigation button is pressed, in which case the home container is hidden and the account container is shown.

Other than that, I've refactored `index.html` to fix inconsistent indentation.

--------

**EDIT:** @thecsw created a quick testing page https://twinproduction.github.io/ghpagetest/

robalb committed:
moved account page to a smaller modal box

robalb commented:
there was already a search bar in the nav bar, and since the data to show in the user page is very small, i just moved it in a simpler modal box

TwiN commented:
@robalb Good idea, hadn't thought of that until I saw the modal that was added for the upcoming economy reset when I resolved the conflicts.

I was also working on that on my end, but you beat me to it.

robalb commented:
ok, i think the account page is ready. In future we'll probably need a separate page to show more data, (like all the active investments with a graph) and have te ability to link directly to it from the reddit bot, so that users don't have to always type in their username

TwiN committed:
Updated README.md

TwiN commented:
@robalb I was thinking of creating another page too, but then I thought about all the redundant code we'd have to deal with (e.g. navbar, scripts, etc.).

Though, we could probably add a parameter (e.g. `index.html?account=twinproduction`) and when the page is finished loading, check if there's a parameter `account` in the url, if there's one, then display the account information.

-----

UPDATE: I'm working on that

robalb commented:
@TwinProduction yeah, but it's probably not worth it. We are planning to integrate a reddit client in the website, to view all the posts in memeconomy in real time. when we'll do it, there will be enough stuff to justify a separate user page with a direct url

TwiN committed:
Automatically show account if url contains ?account=

TwiN commented:
@robalb no worries, for now, I just added a very tiny snippet to do that:

```js
let url = new URL(window.location.href);
let username = url.searchParams.get("account");
if (username) {
    document.getElementById('investor-username').value = username;
    showAccount('investor-username');
    history.pushState(null, '', window.location.href.split('?')[0]);
}
```

It automatically opens up the account modal if the url contains `?account=...` and removes the `account` parameter from the url afterward. 

Live example: https://twinproduction.github.io/ghpagetest/?account=twinproduction

TwiN committed:
Cleaned up empty block (whoops)

robalb committed:
moved code to closure module

robalb commented:
ok, i just cleared a bit the code moving everything related with the user modal into a closure module

robalb committed:
removed useless variable

TwiN commented:
Looks good to me.
Push when you're ready @robalb @thecsw 

There's quite a few changes in this PR, so it's probably better to push it sooner than later or we'll end up with a long list of conflicts.

robalb commented:
uhmm i'm an idiot
`?account=<IMG SRC=/ onerror="alert('pwned bro')"></img>`

TwiN commented:
@robalb did you introduce that xss, or was that me?

I don't remember directly parsing the account parameter o.O

-------

**EDIT:** It's because of the `${username} is not an investor` message that you added. ~~Can you fix it or should I?~~ see https://github.com/MemeInvestor/memeinvestor_bot/pull/78
